include/nnstreamer-edge.h

```diff
@@ -96,119 +96,360 @@ typedef void (*nns_edge_data_destroy_cb) (void *data);
+ * IP or HOST           | IP address of the node to accept connection from other node.
+ * PORT                 | Port of the node to accept connection from other node.
+ * DEST_IP or DEST_HOST | IP address of the destination node.
+ * DEST_PORT            | Port of the destination node
+ * TOPIC                | Topic used to publish/subscribe to/from the broker.
+ * ID or CLIENT_ID      | Unique indentifier of the edge handle or client ID. (Read-only)
  */
 int nns_edge_set_info (nns_edge_h edge_h, const char *key, const char *value);
 
 /**
  * @brief Get nnstreamer edge info.
  * @note The param key is case-insensitive. Caller should release returned string using free().
+ * @param[in] edge_h The edge handle.
+ * @param[in] key Identifiers to determine which value to get.
+ * @param[out] value The values that match the key.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
  */
 int nns_edge_get_info (nns_edge_h edge_h, const char *key, char **value);
 
 /**
  * @brief Get the nnstreamer edge event type.
+ * @param[in] event_h The edge event handle.
+ * @param[out] event event type. value of @a nns_edge_event_e.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid.
  */
 int nns_edge_event_get_type (nns_edge_event_h event_h, nns_edge_event_e *event);
 
 /**
  * @brief Parse edge event (NNS_EDGE_EVENT_NEW_DATA_RECEIVED) and get received data.
  * @note Caller should release returned edge data using nns_edge_data_destroy().
+ * @param[in] event_h The edge event handle.
+ * @param[out] data_h Handle of received data.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid
  */
 int nns_edge_event_parse_new_data (nns_edge_event_h event_h, nns_edge_data_h *data_h);
 
 /**
  * @brief Parse edge event (NNS_EDGE_EVENT_CAPABILITY) and get capability string.
  * @note Caller should release returned string using free().
+ * @param[in] event_h The edge event handle.
+ * @param[out] capability Capability string.
+ * @return 0 on success. Otherwise a negative error value.
+ * @retval #NNS_EDGE_ERROR_NONE Successful.
+ * @retval #NNS_EDGE_ERROR_NOT_SUPPORTED Not supported.
+ * @retval #NNS_EDGE_ERROR_INVALID_PARAMETER Given parameter is invalid
  */
 int nns_edge_event_parse_capability (nns_edge_event_h event_h, char **capability);
 
 /**
  * @brief Create nnstreamer edge data.
+ * @param[out] data_h Handle of edge data. data handle should be released using nns_edge_data_destroy().
```

Comment:
line 359 and 381
`Caller should release returned handle using nns_edge_data_destroy().`